File: WorldFeed/Server/src/WorldFeed.Common/Models/Properties/SearchResults.cs
namespace WorldFeed.Common.Models.Properties
{
    using System.Collections.Generic;
    using System.Linq;

    using global::WorldFeed.Common.Public.Models.Interfaces;
    using global::WorldFeed.Common.Public.Models.Interfaces.DTO;

    public class SearchResults : ISearchResults
    {
        public SearchResults(IEnumerable<ITweetWithSearchMetadata> tweets, ISearchMetadata searchMetadata)
        {
            Tweets = tweets?.ToArray();
            SearchMetadata = searchMetadata;
        }

        public ITweetWithSearchMetadata[] Tweets { get; }

        public ISearchMetadata SearchMetadata { get; }
    }
}

File: Sharebook/Client/src/app/core/Core/Models/Properties/SearchResults.ts
import {ISearchResults} from "../../../Public/Models/Interfaces/ISearchResults";
import {ITweetWithSearchMetadata} from "../../../Public/Models/Interfaces/ITweetWithSearchMetadata";
import {ISearchMetadata} from "../../../Public/Models/Interfaces/DTO/ISearchMetadata";

export class SearchResults implements ISearchResults {
  constructor(tweets: Array<ITweetWithSearchMetadata>, searchMetadata: ISearchMetadata) {
    this.tweets = tweets;
    this.searchMetadata = searchMetadata;
  }

  public tweets: ITweetWithSearchMetadata[];
  public searchMetadata: ISearchMetadata;
}
